src/RouterServer.ts

```diff
@@ -106,12 +114,15 @@ export class RouterServer {
         if (this.server) {
             throw new Error("Already listening.");
         }
-        console.log("Starting server...");
-        this.server = http.createServer(this.requestListener.bind(this));
+        if (this.httpsOptions) {
+            this.server = https.createServer(this.httpsOptions, this.requestListener.bind(this));
+        } else {
+            this.server = http.createServer(this.requestListener.bind(this));
+        }
         this.server.timeout = 10000;
 
         this.server.listen(port, "0.0.0.0", () => {
-            console.log("Server running at http://0.0.0.0:" + port);
+            console.log("Server running at https://0.0.0.0:" + port);
```

Comment:
Make a switch between https and http here

Reply:
fixed